CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+5. Run SyMPC tests
+6. Run PySyft tests
+
+## Code Quality
+
+### Formatting and Linting
+
+We use several tools to keep our codebase high quality.
+
+- black
+- flake8
+- isort
+
+### Tests and CI
+
+When you push your code it will run through a series of GitHub Actions which will ensure that the code meets our minimum standards of code quality before a Pull Request can be reviewed and approved.
+
+To make sure your code will pass these CI checks before you push you should use the pre-commit hooks and run tests locally.
+
+We aim to have a 100% test coverage, and the GitHub Actions CI will fail if the coverage is below a certain value. You can evaluate your coverage using the following commands.
+
+```
+$ pytest -m fast -n auto
+```
+
+### Writing Test Cases
+
+Always make sure to create the necessary tests and keep test coverage at 100%. You can always ask for help in slack or via GitHub if you don't feel confident about your tests.
+
+### Documentation and Code Style Guide
+
+To ensure code quality and make sure other people can understand your changes, you have to document your code. For documentation, we are using the Google Python Style Rules which can be found [here](https://github.com/google/styleguide/blob/gh-pages/pyguide.md). A well-written example can we viewed [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+
+Your documentation should not describe the obvious, but explain what's the intention behind the code and how you tried to realize your intention.
+
+You should also document non-self-explanatory code fragments e.g. complicated for-loops. Again please do not just describe what each line is doing but also explain the idea behind the code fragment and why you decided to use that exact solution.
+
+You can check documentation style errors with:
+
+```bash
+# Check if the description matches a function signature
+$ darglint .
+# Checks compliance with python docstrings conventions
+$ pydocstyle .
+```
+
+### Imports Formatting
+
+We use isort to automatically format the python imports. 
+Run isort manually like this:
+
+```
+$ isort .
+```
+
+### Common Security Issues
+
+Security issues are hard to avoid if you dont know them. To avoid the most common security issues we use bandit.
+We can execute:
+
+```
+$ bandit .
+```
+
+### Generating Documentation
+
+You can execute the following command from the root directory of the project. It will:
+
+1. Navigate to the docs folder
+2. Build the html files
+3. Navigate to the folder with the generated html files
+4. Start a local http server
+
+```
+$ cd docs && make html && cd ../build/sphinx/html && python3 -m http.server
+```
+
+You can now navigate to http://0.0.0.0:8000/.
+
+## Pre-Commit
+
+We are using a tool called [pre-commit](https://pre-commit.com/) which is a plugin system that allows easy configuration of popular code quality tools such as linting, formatting, testing and security checks.
+
+### MacOS
```

Comment:
Remove the install - we do this in `requirements.dev.txt` (we need only the `pre-commit install` I think)